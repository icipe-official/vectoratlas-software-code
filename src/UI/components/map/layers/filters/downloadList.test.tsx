import React from 'react';
import { fireEvent, render } from '../../../../test_config/render';
import {
  drawerListToggle,
  drawerToggle,
  initialState,
} from '../../../../state/map/mapSlice';
import DownloadList from './downloadList';

/* eslint-disable react/display-name*/
jest.mock('@mui/material/ListItemIcon', () => () => (
  <div>ListItemIcon icon</div>
));
jest.mock('@mui/icons-material/ExpandLess', () => () => (
  <div>ExpandLess icon</div>
));
jest.mock('@mui/material/ListItemText', () => (props: any) => (
  <div>List item text mock {JSON.stringify(props)}</div>
));
jest.mock('@mui/material/ListItemButton', () => (props: any) => (
  <div onClick={props.onClick}>
    <div>List item button mock</div>
    {props.children}
  </div>
));

/* eslint-enable react/display-name*/

describe('DownloadList', () => {
  let state: any;

  beforeEach(() => {
    state = { map: initialState() };
  });

  it('renders correctly when collapsed', () => {
    const { wrapper } = render(
      <DownloadList sectionTitle="Download" sectionFlag="download" />,
      state
    );

    expect(wrapper.container).toMatchSnapshot();
  });

  it('renders correctly when expanded', () => {
    state.map.map_drawer.open = true;
    state.map.map_drawer.download = true;

    const { wrapper } = render(
      <DownloadList sectionTitle="Download" sectionFlag="download" />,
      state
    );

    expect(wrapper.container).toMatchSnapshot();
  });

  it('closes just the section if the drawer is open', async () => {
    state.map.map_drawer.open = true;
    state.map.map_drawer.download = true;

    const { wrapper, store } = render(
      <DownloadList sectionTitle="Download" sectionFlag="download" />,
      state
    );

    const sectionButton = await wrapper.findByText('List item button mock');
    fireEvent.click(sectionButton);

    const actions = store.getActions();
    expect(actions).toHaveLength(1);
    expect(actions[0]).toEqual(drawerListToggle('download'));
  });

  it('opens the drawer and the section if the drawer is closed', async () => {
    state.map.map_drawer.open = false;

    const { wrapper, store } = render(
      <DownloadList sectionTitle="Download" sectionFlag="download" />,
      state
    );

    const sectionButton = await wrapper.findByText('List item button mock');
    fireEvent.click(sectionButton);

    const actions = store.getActions();
    expect(actions).toHaveLength(2);
    expect(actions[0]).toEqual(drawerToggle());
    expect(actions[1]).toEqual(drawerListToggle('download'));
  });
});
