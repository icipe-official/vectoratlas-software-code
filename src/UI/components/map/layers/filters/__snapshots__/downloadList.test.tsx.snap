// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadList renders correctly when collapsed 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
  >
    <div>
      <div>
        List item button mock
      </div>
      <div>
        ListItemIcon icon
      </div>
      <div>
        List item text mock 
        {"primary":"Download","sx":{"opacity":0}}
      </div>
    </div>
  </li>
</div>
`;

exports[`DownloadList renders correctly when expanded 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
  >
    <div>
      <div>
        List item button mock
      </div>
      <div>
        ListItemIcon icon
      </div>
      <div>
        List item text mock 
        {"primary":"Download","sx":{"opacity":1}}
      </div>
      <div>
        ExpandLess icon
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pcvi2z-MuiCollapse-root"
      data-testid="downloadListContainer"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="map"
            id="map"
          />
          <a
            class="MuiTypography-root MuiTypography-overline MuiLink-root MuiLink-underlineHover btn btn-outline-dark css-10qtxo4-MuiTypography-root-MuiLink-root"
            id="export-png-draw"
            style="display: flex; justify-content: center;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              DOWNLOAD MAP IMAGE
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            download="map.png"
            id="image-download"
          />
        </div>
      </div>
    </div>
  </li>
</div>
`;
